### ISSUE TITLE
Facebook Messenger Error -- Cannot Send Image/File

### ISSUE THREAD

**Issue by weems** [brave/browser-laptop#698]
1. Open a New Message
2. Click the Image Icon to Send a Image
3. Select the File Name, Click Select
4. Notice No File is Uploaded to the Message

**Comment by luixxiul** [comment]
Still persists?

CC: @bbondy

**Comment by cezaraugusto** [comment]
Confirmed on OS X.

Strange that it only happens when you hit "open" (OS X). If you double click the file it upload as expected.

After double clicking for the first time, either method works as expected.

**Comment by tbass71** [comment]
This also happens in the comments section on Facebook. I am using Windows 7 SP1.

1. Click the Image Icon to Send a Image in the comments section of any post.
2. Select the File Name, Click Select
3. Notice No File is Uploaded to the comments.

**Comment by ktd99** [comment]
Is there a work-around at this point?  I need to be able to attach a File.

**Comment by bsclifton** [comment]
@ktd99 sorry for the long response- Are you able to try to upload via drag and drop? Just drag the image onto the window and it should work

**Comment by ktd99** [comment]
Yes...now able with side-by-side windows!

 

Is that work-around going to become the way we insert files, or will Brave develop a more standard interface for attaching files to a FB msg?

**Comment by bsclifton** [comment]
@ktd99 Ideally, the happy path (being able to pick the file the way you tried) would work; we can keep this issue open to track that.  Super glad that there's a work-around in the meantime

Thanks for hanging in there with us! I know you've been a long time Brave user and we definitely appreciate your patience and feedback 😄

**Comment by bsclifton** [comment]
Moving to 0.13.7

**Comment by ktd99** [comment]
I sure hope we don't revisit old issues.  

 

I know, technically, this is different.  But I can't download a file from the Internet without losing its file type.  It gets downloaded as "all files."   I hate to have to go back to another browser for that basic function.

**Comment by luixxiul** [comment]
That issue was captured here: https://github.com/brave/browser-laptop/issues/2231

**Comment by bsclifton** [comment]
Moving to 0.14.3

**Comment by alexwykoff** [comment]
@darkdh @bbondy thought you might be able to pitch in on this one, should I assign it to you?

**Comment by darkdh** [comment]
I will check it out after #9434

**Comment by darkdh** [comment]
per https://github.com/brave/browser-laptop/issues/698#issuecomment-251121157
it requires 3rd party cookie unblocking which needs different solution than https://github.com/brave/muon/issues/241

**Comment by darkdh** [comment]
Test Plan:
messenger
1. Open a New Message
2. Click the Image Icon to Send a Image
3. Select the File Name, Click Select
4. File/Image should be uploaded

comment
1. Clear your session data
2. Open Brave
3. Go to facebook
4. The shield should show allow all cookies
5. You should be able to upload image to comment

Respect users setting:
1. Make 3rd party cookie block on facebook/google drive
2. Upgrade Brave
3. Go to facebook/google drive
4. The shield should show block 3rd party cookies

**Comment by ktd99** [comment]
There have been SO many issues with Brave that I've had to stop using it as my primary browser.

 

Are the necessary fixes in sight...??

 

        Karen

**Comment by darkdh** [comment]
Hi @ktd99 ,

It will be fixed by https://github.com/brave/muon/pull/241 and https://github.com/brave/browser-laptop/issues/9879

**Comment by bsclifton** [comment]
@ktd99 sorry to hear about all the issues you've ran into... but thanks for giving us a try and helping us collect info needed to fix the bugs.  We're definitely getting better!  Were there any other big issues which led to you switching? (or was it a lot of small bugs?)

**Comment by ktd99** [comment]
The primary issue was the inability to Attach files in Facebook messages as the software is designed to do.  Drag/drop was a pain.

 

I also had to use a number of Google docs shared files...simpler under Google.  Not even sure I could have installed the application in Brave.  (I switched a while ago...don't recall.)

 

Lastly, while engaging in online orders, I found problems adding to the Cart.  The Cart was not updated...like certain commands were being ignored.

 

I'd like to be able to return to Brave at some point...but the functionality needs to be upgraded.
